### PR TITLE
Fixed steep surfaces bug and implemented sliding

### PR DESCRIPTION
**Pull request by mwert09**
Fixed player climbing steep surfaces bug and implemented sliding.

Issue: https://github.com/UnityTechnologies/open-project-1/issues/8
Card: https://open.codecks.io/unity-open-project-1/decks/13/card/16f-player-can-climb-steep-surfaces

### CODE REVIEW

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 1]
```diff
@@ -15,22 +15,32 @@ public class Character : MonoBehaviour
...
     private bool isJumping = false; //If true, a jump is in effect and the player is holding the jump button
     private float jumpBeginTime = -Mathf.Infinity; //Time of the last jump
     private float turnSmoothSpeed; //Used by Mathf.SmoothDampAngle to smoothly rotate the character to their movement direction
     private float verticalMovement = 0f; //Represents how much a player will move vertically in a frame. Affected by gravity * gravityContributionMultiplier
+    private Vector3 hitNormal; // ground normal
+    private bool shouldSlide; // Should player slide?
     private Vector3 inputVector; //Initial input horizontal movement (y == 0f)
     private Vector3 movementVector; //Final movement vector
+    
+
```
Comment (atb-brown) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:31]: Looks like some extra whitespace.
```suggestion
```

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 2]
```diff
@@ -15,22 +15,32 @@ public class Character : MonoBehaviour
...
     {
         characterController = GetComponent<CharacterController>();
+        
     }
 
     private void Update()
     {
+       
+        
+
```
Comment (atb-brown) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:43]: Looks like some extra whitespace.
```suggestion
```

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 3]
```diff
@@ -83,10 +94,23 @@ private void Update()
...
         //Apply the result and move the character in space
         movementVector.y = verticalMovement;
         characterController.Move(movementVector * Time.deltaTime);
 
+        // check if the controller is grounded and above slope limit
+        // if player is grounded and above slope limit
+        // player has to slide
+        shouldSlide = !(Vector3.Angle(Vector3.up, hitNormal) <= characterController.slopeLimit);
```
Comment (HyagoOliveira) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:102]: Maybe add some variables to clean this code part?
Adding more variables is a good way to increase legibility among other developers.

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 4]
```diff
@@ -83,10 +94,23 @@ private void Update()
             }
         }
 
+        // if player has to slide then add sideways speed to make it go down
```
Comment (HyagoOliveira) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs:89]: Maybe add a private function called `UpdateSlide()` ?

File: UOP1_Project/Assets/Scripts/Characters/Character.cs [hunk 5]
```diff
@@ -15,22 +15,32 @@ public class Character : MonoBehaviour
...
 
     private void Awake()
     {
         characterController = GetComponent<CharacterController>();
+        
     }
 
     private void Update()
     {
+
```
Comment (HyagoOliveira) [inline UOP1_Project/Assets/Scripts/Characters/Character.cs]: Why those empty lines?

**Review by HyagoOliveira — COMMENTED** [review]
Added some suggestions that may improve code legibility.

File: UOP1_Project/Assets/Scripts/Characters/Character.cs (hunk 3 above)
Comment (mwert09) [reply UOP1_Project/Assets/Scripts/Characters/Character.cs]: Thank you. I tried to change it a little bit how does it look now? I am just a beginner so i have a lot to learn.

**Comment by mwert09** [comment]
> Added some suggestions that may improve code legibility.

How does it look now?

File: UOP1_Project/Assets/Scripts/Characters/Character.cs (hunk 3 above)
Comment (HyagoOliveira) [reply UOP1_Project/Assets/Scripts/Characters/Character.cs]: Sometimes a good variable or method name can replace a lot of comments. Also, it indicates you care about legibility which is always good.

**Comment by ciro-unity** [comment]
I like this solution, it gets in just a few lines of code and the control of the character stays good.

However, I think it has a few issues now when you hit your head under the floating platform, after I merged in `master` some changes that were fixing another issue. So now the character literally bounces when hitting their head, which can cause some weird acceleration downwards.

Can you please try taking a look at that? If you can tweak that I'd gladly accept this as a fix.
Thanks!

**Comment by mwert09** [comment]
> I like this solution, it gets in just a few lines of code and the control of the character stays good.
> 
> However, I think it has a few issues now when you hit your head under the floating platform, after I merged in `master` some changes that were fixing another issue. So now the character literally bounces when hitting their head, which can cause some weird acceleration downwards.
> 
> Can you please try taking a look at that? If you can tweak that I'd gladly accept this as a fix.
> Thanks!

I think it works now. If there is an issue please let me know. I am very curious. Will I be able to see my name on end credits. That would be awesome!

**Comment by ciro-unity** [comment]
> Will I be able to see my name on end credits?
Apparently yes :D

**Pull request merged**